tests/numerics/wave_solver_hermite.cc

```diff
@@ -0,0 +1,425 @@
+  max_error = l2_errors[0];
+
+  // Make an initial time-step using
+  // u(t+dt) = u(t) + dt u_t(t) +
+  //              0.5 dt^2 M^(-1)A u(t) + O(dt^3, dt^2 dx^(p+1))
+  tstep_noninvert.vmult(temp, sol_curr);
+  mass.vmult(sol_next, sol_prev);
+  sol_next *= dt;
+  temp += sol_next;
+  sol_next = 0;
+
+  wave.update_time(dt);
+  VectorTools::project_hermite_boundary_values(
+    mapping_h,
+    dof,
+    boundary_functions,
+    face_quadrature,
+    VectorTools::HermiteBoundaryType::hermite_dirichlet,
+    boundary_values);
+
+  mass_solve.copy_from(mass);
+  MatrixTools::apply_boundary_values(
+    boundary_values, mass_solve, sol_next, temp, true);
+
+  SparseDirectUMFPACK mass_inv;
+  mass_inv.initialize(mass_solve);
+  mass_inv.vmult(sol_next, temp);
+
+  sol_prev = sol_curr;
+  sol_curr = sol_next;
+  sol_next = 0;
+
+  boundary_values.clear();
+
+  l2_errors.emplace_back(
+    l2_error<dim>(mapping_h, dof, quadrature, wave, sol_curr));
+  max_error = std::max(max_error, *l2_errors.cend());
+
+  while (wave.get_time() < final_time)
+    {
+      mass.vmult(sol_next, sol_prev);
+      tstep_noninvert.vmult(temp, sol_curr);
+      temp *= 2.0;
+      temp -= sol_next;
+      sol_next = 0;
+
+      wave.update_time(dt);
+      VectorTools::project_hermite_boundary_values(
+        mapping_h,
+        dof,
+        boundary_functions,
+        face_quadrature,
+        VectorTools::HermiteBoundaryType::hermite_dirichlet,
+        boundary_values);
+
+      mass_solve.copy_from(mass);
+      MatrixTools::apply_boundary_values(
+        boundary_values, mass_solve, sol_next, temp, true);
+      mass_inv.vmult(sol_next, temp);
+
+      sol_prev = sol_curr;
+      sol_curr = sol_next;
+      sol_next = 0;
+      boundary_values.clear();
+
+      l2_errors.emplace_back(
+        l2_error<dim>(mapping_h, dof, quadrature, wave, sol_curr));
+      max_error = std::max(max_error, *(--l2_errors.cend()));
+    }
+
+  deallog << std::endl;
+  char fname[50];
+  sprintf(fname, "Cell-%dd-Hermite-%d", dim, regularity);
+  deallog.push(fname);
+
+  deallog << "initial_time: " << initial_time << std::endl;
+  deallog << "Final time: " << final_time << std::endl;
+  deallog << std::endl;
+
+  deallog << "Error at final time: " << *(--l2_errors.cend()) << std::endl;
+  deallog << "Largest error: " << max_error << std::endl;
+  deallog << std::endl;
+
+  double t = initial_time - dt;
+  deallog << "Error time series:";
+  for (const auto &it : l2_errors)
+    deallog << std::endl << (t += dt) << "\t" << it;
+  deallog << "\n\n" << std::endl;
+
+  deallog.pop();
+}
+
+
+
+int
+main()
+{
+  std::ofstream logfile("output");
+  deallog << std::setprecision(8) << std::fixed;
+  deallog.attach(logfile);
+
+  test_wave_solver<1>(-0.3, 1);
+
+  test_wave_solver<1>(0.0, 1);
+  test_wave_solver<1>(0.0, 2);
+  test_wave_solver<1>(0.0, 3);
+
+  test_wave_solver<2>(0.0, 1);
+  test_wave_solver<2>(0.0, 2);
+  test_wave_solver<2>(0.0, 3);
+
+  test_wave_solver<3>(0.0, 1);
```

Comment:
I am wondering how long this test is running? If it takes too long, we could consider to decrease the number of timesteps.